homeassistant/components/light/__init__.py

```diff
@@ -241,6 +241,16 @@ def async_handle_light_service(service):
         if color_name is not None:
             params[ATTR_RGB_COLOR] = color_util.color_name_to_rgb(color_name)
 
+        xy_color = params.get(ATTR_XY_COLOR, None)
+        brightness = params.get(ATTR_BRIGHTNESS, None)
+
+        # Convert xy_color/brightness for lights that do not support xy
+        if xy_color is not None and brightness is not None:
```

Comment:
I added commits so it now works like that. It is a bit ugly because we can have mixed lights.